Community/Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js

```diff
@@ -0,0 +1,400 @@
+// All credit for original plugin logic goes to Migz.
+// This Plugin is essentially just his NVENC/CPU plugin modified to work with QSV & with extra hevc logic.
+// Extra logic is mainly to control encoder quality/speed & to allow HEVC files to be reprocessed to reduce file size
+// NOTE - This does not use VAAPI, it is QSV only. So newer intel igpus only. 8th+ gen should work.
+function details() {
+  return {
+    id: 'Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC',
+    Stage: 'Pre-processing',
+    Name: 'Boosh-Transcode using QSV GPU & FFMPEG',
+    Type: 'Video',
+    Operation: 'Transcode',
+    Description: `This is a QSV specific plugin, VAPPI is NOT used. So an INTEL QSV enabled CPU is required. 
+    8th+ gen should work. Files not in H265/HEVC will be transcoded into H265/HEVC using Quick Sync Video (QSV) 
+    via Intel GPU with ffmpeg. Settings are dependant on file bitrate working by the logic that H265 can support 
+    the same ammount of data at half the bitrate of H264. This plugin will skip files already in HEVC, AV1 & VP9 
+    unless "reconvert_hevc" is marked as true. If it is then these will be reconverted again into hevc if they 
+    exceed the bitrate specified in "hevc_max_bitrate". Reminder! An INTEL QSV enabled CPU is required.`,
+    Version: '1.0',
+    Tags: 'pre-processing,ffmpeg,video only,qsv,h265,hevc,configurable',
+    Inputs: [{
+      name: 'container',
+      type: 'string',
+      defaultValue: 'mkv',
+      inputUI: {
+        type: 'dropdown',
+        options: [
+          'mkv',
+          'mp4',
+        ],
+      },
+      tooltip: `Specifies the output container of the file
+      \\n Ensure that all stream types you may have are supported by your chosen container.
+      \\n MKV is recommended.
+      \\nExample:\\n
+      mkv
+      \\nExample:\\n
+      mp4`,
+    },
+    {
+      name: 'encoder_speedpreset',
+      type: 'string',
+      defaultValue: 'slower',
```

Comment:
Not sure about this, but does `slower` provide any significant savings/quality compared to `slow`?

Reply:
Hard to say. Intel did a white paper regarding QSV on ffmpeg performance. - https://www.intel.com/content/dam/www/public/us/en/documents/white-papers/cloud-computing-quicksync-video-ffmpeg-white-paper.pdf
They describe `slower` as close to best quality with faster performance. `Slow` is just described as between `slower` and `medium`.
Regardless I've defaulted this to `slow` now since I suspect that's the profile most would prefer, and the user can change this if they want to. Also added this link into the script so user can view it and make their own decision.